src/utils/clients.py

```
@@ -306,14 +327,20 @@ async def _call_with_provider_selection() -> (
     if enable_retry:
 
         def before_retry_callback(retry_state: Any) -> None:
```

Ruff (0.14.7)

Dynamically typed expressions (typing.Any) are disallowed in `retry_state`

(ANN401)